### ISSUE TITLE
Explicit schema versions

### ISSUE THREAD

**Issue by kristiandupont** [realm/realm-dotnet#518]

**Comment by AndyDentFree** [comment]
Is this needed to make migrations robust and especially fix issues with trivial migration?

**Comment by fealebenpae** [comment]
The migration callbacks implement their business logic conditionally based on the schema version numbers. We already have the API needed for users to set the schema version, we should just encourage its use, IMO.

**Comment by AndyDentFree** [comment]
We have already, and recently, confirmed these work for some SO questions and help requests. I propose a quick check to confirm initial value and then add some main documentation will be enough.

**Comment by AndyDentFree** [comment]
We set our initial version in `RealmConfiguration`:

```
// to match ObjectStore::NotVersioned
 public static UInt64 NotVersioned => UInt64.MaxValue;

...
public UInt64 SchemaVersion { get; set;} = RealmConfiguration.NotVersioned;
```

However, that was a misunderstanding of how  Cocoa uses the ObjectStore `std::numeric_limits<uint64_t>::max`. We should be using zero to match Cocoa and Java.

**Comment by fealebenpae** [comment]
Implemented in #768
